### ISSUE TITLE
Create an Aurelia-CLI config reference library (cookbook)

### ISSUE THREAD

**Issue by Thanood** [monterey-framework/monterey#394]
### Intention

**I'm submitting a feature request**  

### Environment

* **Operating System:**
Windows 10

* **Node Version:**
v6.2.2

* **NPM Version:**
3.10.5

### Description
**Current situation**:

I'm seeing a lot of questions in Aurelia-CLI issue tracker on how to change the build scripts or any other kind of configuration. For example "how to let BrowserSync proxy requests" or "how to enable push-state debugging with BrowserSync".

Most of the time, the answer seems to be "what's stopping you from changing the build tasks?".
Frankly, this is not an answer at all.

**Expected/desired behavior:** 

Monterey should have a cookbook repository as a *companion site* (because you can't link from GitHub issues to a desktop app).

Monterey itself should support this repository by providing a means to easily share a *recipe* like f.i. a BrowserSync proxy configuration.

**What is the motivation / use case for changing the behavior?**  

To be able to answer questions like this with:
"This feature request is covered by changing the build tasks. Have a look at our cookbook (link to Monterey cookbook repository site). There are a few recipes that may help you."

**Comment by Thanood** [comment]
A few first recipes:

### Enable BrowserSync to proxy a backend server

In `aurelia_project/tasks/run.js` change the `server` task to:
```javascript
let serve = gulp.series(
  build,
  done => {
    browserSync({
      // online: false,
      // notify: true,
      open: false,
      port: 9000,
      logLevel: 'silent',
      proxy: {
        target: 'localhost:24562'
      }
    }, function(err, bs) {
      let urls = bs.options.get('urls').toJS();
      console.log(`Application Available At: ${urls.local}`);
      console.log(`BrowserSync Available At: ${urls.ui}`);
      done();
    });
  }
);
```

Replace `localhost:24562` to the url you'd like to proxy.

###  (*should work but doesn't*) Enable requirejs loading from 'scripts' directory

Add this to `build.loader` in `aurelia.json`:
```json
"config": {
  "baseUrl": "scripts/"
},
``´

**Comment by JeroenVinke** [comment]
I like this idea, but perhaps it's better to keep this in the original aurelia-cli repository as documentation, assuming that the aurelia-cli folks like this idea 😄 Maybe it would even be visible on the aurelia hub

**Comment by Thanood** [comment]
Yes, I think you're right.. I could issue a PR when I understand their doc structure.. :smiley:
